src/features/courses/course-route-validation.ts

```diff
@@ -467,3 +467,22 @@ export const listAttachmentsValidation = {
     },
     body: {},
 };
+
+export const emailProfValidation = {
+    params: {
+        id: Joi.number().required(),
+    },
+    query: {},
+    body: {
+        content: Joi.string().required(),
+        topic: Joi.object({
+            id: Joi.number().required(),
+            name: Joi.string().required(),
+        }).required(),
+        question: Joi.object({
+            id: Joi.number().required(),
+            webworkQuestionPath: Joi.string().required(),
```

Comment:
We should remove this

Reply:
Problem number as well, we should be getting this from a fetched database object, otherwise a malicious user could spoof easily

Yes we could add permission checks but then we will be running the same queries anyway

Reply:
Where does the problemNumber change when randomizeOrder is true?

Reply:
I guess the problemId doesn't change, and the user doesn't see that, so it should work regardless.

Reply:
I wouldn't really think there's much for a malicious user to spoof here. We could lock it down and always do the database calls, but the important information- the user initiating the request- is gotten from the session.

Reply:
I mean even if it's not a spoof, I think the database queries have to occur anyway, that will always be 100% up to date as well as avoid any bugs on the frontend (like state management bugs)